### PR TITLE
Add topic guide for s3 configs

### PR DESCRIPTION
**Pull request by jamesls**
Documents the config values added in #1122.

cc @kyleknap @danielgtaylor

### CODE REVIEW

File: awscli/topics/s3-config.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
+:title: AWS CLI S3 Configuration
+:description: Advanced configuration for AWS S3 Commands
+:category: S3
+:related command: s3 cp, s3 sync, s3 mv, s3 rm
+
+The ``aws s3`` transfer commands, which include the ``cp``, ``sync``, ``mv``,
+and ``rm`` commands, have additional configuration you can use to control
```
Comment (kyleknap) [inline awscli/topics/s3-config.rst]: additional configuration ---> additional configuration parameters

File: awscli/topics/s3-config.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Configuration Values
+====================
+
+These are the configuration values you can set for S3:
+
+* ``max_concurrent_requests`` - The maximum number of concurrent requests.
+* ``max_queue_size`` - The maximum number of tasks in the task queue.
+* ``multipart_threshold`` - The size threshold where the CLI uses multipart
+  transfers.
```
Comment (kyleknap) [inline awscli/topics/s3-config.rst]: transfers for individual files.

File: awscli/topics/s3-config.rst [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+
+Configuration Values
+====================
+
+These are the configuration values you can set for S3:
+
+* ``max_concurrent_requests`` - The maximum number of concurrent requests.
+* ``max_queue_size`` - The maximum number of tasks in the task queue.
+* ``multipart_threshold`` - The size threshold where the CLI uses multipart
```
Comment (kyleknap) [inline awscli/topics/s3-config.rst]: threshold where the ---> threshold the

File: awscli/topics/s3-config.rst [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  transfers.
+* ``multipart_chunksize`` - When using multipart transfers, this is the chunk
```
Comment (kyleknap) [inline awscli/topics/s3-config.rst]: For consistency:

The chunk size the CLI uses for multipart transfers of individual files

File: awscli/topics/s3-config.rst [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  S3 transfer.
+
+
+max_queue_size
+--------------
+
+**Default** - ``1000``
+
+The AWS CLI internally uses a producer consumer model, where we queue up S3
+tasks that are then executed by consumers, which in this case utilize a bound
```
Comment (kyleknap) [inline awscli/topics/s3-config.rst]: You might want to explain what a task is.

File: awscli/topics/s3-config.rst [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+When uploading, downloading, or copying a file, the S3 commands
+will switch to multipart operations if the file reaches a given
+size threshold.  The ``multipart_threshold`` controls this value.
+You can specify this value in one of two ways:
+
+* The file size in bytes.  For example, ``1048576``.
+* The file size with a size suffix.  You can use ``KB``, ``MB``, ``GB``,
+  ``TB``.  For example: ``10MB``, ``1GB``.  Note that S3 imposes
+  constraints on valid values that can be used for multipart
+  operations.
```
Comment (kyleknap) [inline awscli/topics/s3-config.rst]: Might want to mention that there is a general maximum of a non multipart upload size of 5 GB.

**Comment by kyleknap** [comment]
It looks good. I would say a general comment is that in some places terminology is used like "task" that is not really defined anywhere. It would be nice if a sentence was given to introduce the term (i.e. what it encompasses)

Also can you run the ``scripts/make-topic-index`` script to regenerate the topic index to include the topic?

**Comment by jamesls** [comment]
@kyleknap I believe I've incorporated all the feedback.  Let me know if I missed anything.

**Comment by kyleknap** [comment]
Looks good. Can you generate the topic-tags.json for your new topic as well?  Otherwise, :ship:

**Pull request merged**